packages/ember-views/lib/system/utils.js

```diff
@@ -88,27 +88,35 @@ export function getChildViews(view) {
 }
 
 export function initChildViews(view) {
-  view[CHILD_VIEW_IDS] = [];
+  let childViews = [];
+  CHILD_VIEW_IDS.set(view, childViews);
+  return childViews;
 }
 
 export function addChildView(parent, child) {
-  parent[CHILD_VIEW_IDS].push(getViewId(child));
+  let childViews = CHILD_VIEW_IDS.get(parent);
+  if (childViews === undefined) {
+    childViews = initChildViews(parent);
+  }
+
+  childViews.push(getViewId(child));
 }
 
 export function collectChildViews(view, registry) {
   let ids = [];
   let views = [];
+  let childViews = CHILD_VIEW_IDS.get(view);
 
-  view[CHILD_VIEW_IDS].forEach(id => {
-    let view = registry[id];
+  if (childViews) {
+    (CHILD_VIEW_IDS.get(view) || []).forEach(id => {
```

Comment:
I don't think you need to query the weakmap again here, you can reliably do `childViews.forEach` directly (due to the guard

Reply:
Shoot - you're totally right. I missed that.